packages/serve-runtime/src/useForwardHeaders.ts

```diff
@@ -0,0 +1,25 @@
+import { MeshServePlugin } from './types';
+
+export interface ForwardHeadersPluginOptions {
+  headerNames: string[];
+}
+
+export function useForwardHeaders(headerNames: string[]): MeshServePlugin {
+  return {
+    onFetch({ options, setOptions, context }) {
+      const headers = {
+        ...options.headers,
+      };
+      for (const headerName of headerNames) {
+        const headerValue = context.headers[headerName];
+        if (headerValue) {
+          headers[headerName] = headerValue;
+        }
+      }
+      setOptions({
+        ...options,
+        headers,
+      });
```

Comment:
I do think that the priority should be given to options. Otherwise, you can't override the headers coming from the request.
```suggestion
      const headers = {};
      for (const headerName of headerNames) {
        const headerValue = context.headers[headerName];
        if (headerValue) {
          headers[headerName] = headerValue;
        }
      }
      setOptions({
        ...options,
        headers: {
          ...options.headers,
        },
      });
```

Reply:
You're right! Done with tests!